pkg/api/licenses.go

```diff
@@ -422,170 +483,238 @@ func UpdateLicense(c *gin.Context) {
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "Flag",
-			OldValue:     strconv.FormatInt(oldlicense.Flag, 10),
-			UpdatedValue: strconv.FormatInt(license.Flag, 10),
+			OldValue:     &oldVal,
+			UpdatedValue: &newVal,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.Notes != license.Notes {
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "Notes",
-			OldValue:     oldlicense.Notes,
-			UpdatedValue: license.Notes,
+			OldValue:     &oldlicense.Notes,
+			UpdatedValue: &license.Notes,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.DetectorType != license.DetectorType {
+		oldVal := strconv.FormatInt(oldlicense.DetectorType, 10)
+		newVal := strconv.FormatInt(license.DetectorType, 10)
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "DetectorType",
-			OldValue:     strconv.FormatInt(oldlicense.DetectorType, 10),
-			UpdatedValue: strconv.FormatInt(license.DetectorType, 10),
+			OldValue:     &oldVal,
+			UpdatedValue: &newVal,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.Source != license.Source {
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "Source",
-			OldValue:     oldlicense.Source,
-			UpdatedValue: license.Source,
+			OldValue:     &oldlicense.Source,
+			UpdatedValue: &license.Source,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.SpdxId != license.SpdxId {
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "SpdxId",
-			OldValue:     oldlicense.SpdxId,
-			UpdatedValue: license.SpdxId,
+			OldValue:     &oldlicense.SpdxId,
+			UpdatedValue: &license.SpdxId,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.Risk != license.Risk {
+		oldVal := strconv.FormatInt(oldlicense.Risk, 10)
+		newVal := strconv.FormatInt(license.Risk, 10)
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "Risk",
-			OldValue:     strconv.FormatInt(oldlicense.Risk, 10),
-			UpdatedValue: strconv.FormatInt(license.Risk, 10),
+			OldValue:     &oldVal,
+			UpdatedValue: &newVal,
 		}
 		db.DB.Create(&change)
 	}
 	if oldlicense.Marydone != license.Marydone {
+		oldVal := strconv.FormatBool(oldlicense.Marydone)
+		newVal := strconv.FormatBool(license.Marydone)
 		change := models.ChangeLog{
 			AuditId:      audit.Id,
 			Field:        "Marydone",
-			OldValue:     strconv.FormatBool(oldlicense.Marydone),
-			UpdatedValue: strconv.FormatBool(license.Marydone),
+			OldValue:     &oldVal,
+			UpdatedValue: &newVal,
+		}
+		db.DB.Create(&change)
+	}
+	if (oldlicense.ExternalRef.Data().InternalComment == nil && license.ExternalRef.Data().InternalComment != nil) ||
+		(oldlicense.ExternalRef.Data().InternalComment != nil && license.ExternalRef.Data().InternalComment == nil) ||
+		((oldlicense.ExternalRef.Data().InternalComment != nil && license.ExternalRef.Data().InternalComment != nil) && (*oldlicense.ExternalRef.Data().InternalComment != *license.ExternalRef.Data().InternalComment)) {
```

Comment:
Can it be simplified as following @deo002 as `oldlicense.ExternalRef.Data().InternalComment != license.ExternalRef.Data().InternalComment` will be true if there is a change from `nil` and the second condition will capture the cases where both values are null or same string.

```suggestion
	if (oldlicense.ExternalRef.Data().InternalComment != license.ExternalRef.Data().InternalComment) ||
		((oldlicense.ExternalRef.Data().InternalComment != nil && license.ExternalRef.Data().InternalComment != nil) && (*oldlicense.ExternalRef.Data().InternalComment != *license.ExternalRef.Data().InternalComment)) {
```

Reply:
This will not work if one is nil and the other isn't.